docs/source/tutorial/tutorial.advanced-topics.md

```diff
@@ -367,6 +367,79 @@ await runner.task  # This is not needed in a notebook environment!
 timer.result()
 ```
 
+## Custom parallelization using coroutines
+
+Adaptive by itself does not implement a way of sharing partial results between function executions.
+Its implementation of parallel computation using executors is minimal by design.
+Instead the appropriate way to implement custom parallelization is by using coroutines (asynchronous functions).
+
+We illustrate this approach by using `dask.distributed` for parallel computations in part because it supports asynchronous operation out-of-the-box.
+Let us consider the computation below which has a slow but reusable part, and a fast part that cannot be reused.
+
+```{code-cell} ipython3
+import time
+
```

Comment:
Please re-add the definition of `f` here: without it, the user won't know the goal and the computation is not fully defined.